check_api/src/main/java/com/google/errorprone/util/ASTHelpers.java

```diff
@@ -1724,13 +1724,15 @@ public static TargetType targetType(VisitorState state) {
 
     Type type = new TargetTypeVisitor(current, state, parent).visit(parent.getLeaf(), null);
     if (type == null) {
-      if (CONSTANT_CASE_LABEL_TREE != null
+      Tree actualTree = null;
+      if (YIELD_TREE != null && YIELD_TREE.isAssignableFrom(parent.getLeaf().getClass())) {
+        actualTree = parent.getParentPath().getParentPath().getParentPath().getLeaf();
+      } else if (CONSTANT_CASE_LABEL_TREE != null
           && CONSTANT_CASE_LABEL_TREE.isAssignableFrom(parent.getLeaf().getClass())) {
```

Comment:
There should be a better way of handling these. 😬

Reply:
Agree, but I think reflection is the best option we have right now.

I've been wondering if we should try to use mrjars to avoid some of the reflection, I filed an issue to track that idea: https://github.com/google/error-prone/issues/3756

Reply:
That's a good idea! Thanks for filing the issue. 🙇